gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/service/tasks/LaunchConfigTask.java

```diff
@@ -219,6 +221,14 @@ public final void createConfig() throws IOException {
                                 javaVersion.get().compareTo(JavaVersion.toVersion("15")) < 0
                                         ? disableBiasedLocking
                                         : ImmutableList.of())
+                        // https://bugs.openjdk.org/browse/JDK-8281181 stopped respecting cpu.shares for
+                        // processor count. UseContainerCpuShares can be enabled for the time being, however it
+                        // is deprecated in jdk19 and expired in jdk21: https://bugs.openjdk.org/browse/JDK-8282684
+                        .addAllJvmOpts(
+                                javaVersion.get().compareTo(JavaVersion.toVersion("11")) >= 0
+                                                && javaVersion.get().compareTo(JavaVersion.toVersion("20")) <= 0
```

Comment:
Should we only add `-XX:+UseContainerCpuShares` for JDK [11, 19] as JDK 20.x obsoletes the flag and will accept it as an argument without throwing, but it won't have any effect as it [has already been removed from jdk-20+12](https://github.com/openjdk/jdk/commit/db772276848f6ad2d4d13e892bcd0eb3123d030f) from the code outside https://github.com/openjdk/jdk/blob/jdk-20+26/src/hotspot/share/runtime/arguments.cpp#L552

```suggestion
                                                && javaVersion.get().compareTo(JavaVersion.toVersion("19")) <= 0
```

```
  { "UseContainerCpuShares",        JDK_Version::jdk(19), JDK_Version::jdk(20), JDK_Version::jdk(21) },
```

>      DEPRECATED: An option that is supported, but a warning is printed to let the user know that
>                  support may be removed in the future. Both regular and aliased options may be
>                  deprecated.
>   
>                  Add a deprecation warning for an option (or alias) by adding an entry in the
>                  "special_jvm_flags" table and setting the "deprecated_in" field.
>                  Often an option "deprecated" in one major release will
>                  be made "obsolete" in the next. In this case the entry should also have its
>                  "obsolete_in" field set.
>   
>        OBSOLETE: An option that has been removed (and deleted from globals.hpp), but is still accepted
>                  on the command line. A warning is printed to let the user know that option might not
>                  be accepted in the future.
>   
>                  Add an obsolete warning for an option by adding an entry in the "special_jvm_flags"
>                  table and setting the "obsolete_in" field.
>   
>         EXPIRED: A deprecated or obsolete option that has an "accept_until" version less than or equal
>                  to the current JDK version. The system will flatly refuse to admit the existence of
>                  the flag. This allows a flag to die automatically over JDK releases.
>   
>                  Note that manual cleanup of expired options should be done at major JDK version upgrades:
>                     - Newly expired options should be removed from the special_jvm_flags and aliased_jvm_flags tables.
>                     - Newly obsolete or expired deprecated options should have their global variable
>                       definitions removed (from globals.hpp, etc) and related implementations removed.

Reply:
Good catch, thanks!